assets/src/containers/AssignmentPlanning.js

```diff
@@ -65,130 +60,83 @@ const assignmentTable = assignmentData => {
 function AssignmentPlanning (props) {
   const { classes, disabled, courseId } = props
   if (disabled) return (<Error>Assignment view is hidden for this course.</Error>)
-  const [loaded, error, assignmentDefaultData] = useUserSettingData(courseId, 'assignment')
 
-  const currentSetting = 'My current setting'
-  const rememberSetting = 'Remember my setting'
-  const settingNotUpdated = 'Setting not updated'
-
-  // initial default value that we get from backend and updated value when user save the default setting
-  const [defaultValue, setDefaultValue] = useState('')
-  // assignment data and weight controller
-  const [assignmentFilter, setAssignmentFilter] = useState('')
-  const [assignmentData, setAssignmentData] = useState('')
-  // defaults setting controllers
-  const [defaultCheckboxState, setDefaultCheckedState] = useState(true)
+  const [showSaveSetting, setShowSaveSetting] = useState(false)
+  const [saveSettingClicked, setSaveSettingClicked] = useState(false)
+
+  // this is the filter setting currently set
+  const [assignmentGradeFilter, setAssignmentGradeFilter] = useState(0)
```

Comment:
```suggestion
  const [assignmentGradeFilter, setAssignmentGradeFilter] = useState('')
```
If user has not set any default at all the UI is just showing the spinner. the API call `http://localhost:5001/api/v1/courses/295532/get_user_default_selection?default_type=assignment` response in this case is {"default":""} L83-L85 correctly identifies this case but since the default in `useState(0)` and again we are setting with same 0 values with `setAssignmentGradeFilter(0)` so React Dom is not updating since it doesn't see any change is made. 

To replicated this try to delete the user default setting in the `user_default_selection` or pick a new user you haven't tesing add the user from the admin interface and become that user